src/materials/NEMLStress.C

```diff
@@ -28,143 +31,132 @@ NEMLStress::validParams()
   params.addRequiredParam<FileName>("database", "Path to NEML XML database.");
   params.addRequiredParam<std::string>("model", "Model name in NEML database.");
   params.addParam<std::vector<std::string>>("neml_variable_iname",
-                                            "Names of NEML xml name/value pairs");
-
-  params.addParam<Real>("neml_variable_value0",
-                        "NEML xml variable value for neml_variable_iname[0]");
-  params.addParam<Real>("neml_variable_value1",
-                        "NEML xml variable value for neml_variable_iname[1]");
-  params.addParam<Real>("neml_variable_value2",
-                        "NEML xml variable value for neml_variable_iname[2]");
-  params.addParam<Real>("neml_variable_value3",
-                        "NEML xml variable value for neml_variable_iname[3]");
-  params.addParam<Real>("neml_variable_value4",
-                        "NEML xml variable value for neml_variable_iname[4]");
-  params.addParam<Real>("neml_variable_value5",
-                        "NEML xml variable value for neml_variable_iname[5]");
-  params.addParam<Real>("neml_variable_value6",
-                        "NEML xml variable value for neml_variable_iname[6]");
-  params.addParam<Real>("neml_variable_value7",
-                        "NEML xml variable value for neml_variable_iname[7]");
-
+                                            "Names of NEML XML name/value pairs");
+  params.addParam<std::vector<Real>>("neml_variable_value",
+                                     "Corresponding NEML XML variable values");
+  for (size_t i = 0; i < _nvars_max; ++i)
+  {
+    auto istr = Moose::stringify(i);
+    params.addParam<Real>("neml_variable_value" + istr,
```

Comment:
Maybe we don't even need these `neml_variable_valueX` parameters anymore. If I remember correctly, the only reason we added those was for running stochastic analyses, but now we have the ability to set components of a vector parameter.

Regardless, we'll need to deprecate them and keep them for a while, but maybe we can do a follow-on commit to do that once we verify that the new capability works here.

Reply:
oh, ok, if we can control vector entries these can be set to deprecated

Reply:
Pushed the deprecation up, but it seems I got the syntax for modifying vector entries wrong, so this will fail that one test.

Reply:
Fixed with help from Wen!